src/sentry/eventstream/kafka.py

```diff
@@ -60,19 +60,8 @@ def delivery_callback(self, error, message):
         if error is not None:
             logger.warning('Could not publish message (error: %s): %r', error, message)
 
-    def _send(self, project_id, _type, extra_data=()):
+    def _send(self, project_id, _type, extra_data=(), asynchronous=True):
         assert isinstance(extra_data, tuple)
-
-        # Polling the producer is required to ensure callbacks are fired. This
-        # means that the latency between a message being delivered (or failing
-        # to be delivered) and the corresponding callback being fired is
-        # roughly the same as the duration of time that passes between publish
-        # calls. If this ends up being too high, the publisher should be moved
-        # into a background thread that can poll more frequently without
-        # interfering with request handling. (This does `poll` does not act as
-        # a heartbeat for the purposes of any sort of session expiration.)
-        self.producer.poll(0.0)
```

Comment:
I feel like there may have been a reason that `poll` is called before the `produce` call, but I can't remember exactly what it would have been. It might have just been to imply that the `poll` call would likely not be firing callbacks for the `produce` in the same invocation of `_send` here?